volunteer_planner/settings/sentry.py

```diff
@@ -0,0 +1,42 @@
+"""
+Imports and initializes sentry for usage within this Django instance.
+
+For successful initialization 'sentry-sdk' needs to be installed and some
+environment variables need to be present/set:
+- SENTRY_DSN: sentry data source name (https://docs.sentry.io/product/sentry-basics/dsn-explainer/)  # noqa: E501
+              mandatory, without no initialization will be successfully finished.
+- SENTRY_ENVIRON: string, used as 'environment' tag value
+                  default: 'production'
+- SENTRY_TRACE_RATE: float, (0..1), probability a transaction is performance monitored
+                     default: 0.0
+- SENTRY_SEND_PII: bool, (https://docs.sentry.io/platforms/python/configuration/options/#send-default-pii)  # noqa: 501
+                   default: False
+"""
+import logging
+import os
+
+from distutils.util import strtobool
+
+logger = logging.getLogger(__name__)
+
+try:
+    import sentry_sdk
+    from sentry_sdk.integrations.django import DjangoIntegration
+    from sentry_sdk.integrations.logging import LoggingIntegration
+
+    sentry_sdk.init(
+        dsn=os.environ["SENTRY_DSN"],
+        environment=os.environ.get("SENTRY_ENVIRON", "production"),
+        integrations=[
+            DjangoIntegration(),
+            LoggingIntegration(level=logging.INFO, event_level=logging.ERROR),
+        ],
+        traces_sample_rate=float(os.environ.get("SENTRY_TRACE_RATE", "0.0")),
+        # TODO think - really, I'm serious! - about some good 'traces_sampler'
+        #  implementation and replace 'traces_sample_rate'
+        send_default_pii=strtobool(os.environ.get("SENTRY_SEND_PII", "False")),
+    )
+except (ModuleNotFoundError, ImportError):
+    logger.warning("sentry not installed - skipping", exc_info=True)
+except (KeyError, NameError, ValueError):
```

Comment:
Catch _all_ exceptions here?

Reply:
I see two cases: 
- incomplete installation (module loading fails), or 
- configuration not ready to actively use sentry

The former I'd warn admin about, because it shouldn't happen - although we can continue to work normally.
The latter might be something related to demo or testing stage or an offshore installation, not (yet?) connected to a sentry instance. Not necessarily a warning, but worth some low level information (aka debug).

Reply:
I only meant to catch all in the second (last) except case, so that there can't be any unhandled exception left:

```suggestion
except:
```

Reply:
I was guided by `PEP 8: E722 do not use bare 'except'` ... but am willing to broaden except clause 😁

Reply:
fair enough.